tfutils/base.py

```diff
@@ -198,15 +222,17 @@ def __init__(self,
                 lv = save_params[_k]
             setattr(self, _k, sv)
             setattr(self, 'load_' + _k, lv)
-        self.sameloc = all([getattr(self, _k) == getattr(self, 'load_' + _k) for _k in location_variables])
-        if self.no_save:
-            self.sameloc = False
+        self.sameloc = all([getattr(self, _k) == getattr(
+            self, 'load_' + _k) for _k in location_variables])
+        if 'query' in load_params and not load_params['query'] is None and 'exp_id' in load_params['query']:
+            self.sameloc = self.sameloc & (
+                load_params['query']['exp_id'] == self.exp_id)
 
         for _k in ['do_save', 'save_metrics_freq', 'save_valid_freq', 'cache_filters_freq',
                    'save_filters_freq', 'save_initial_filters', 'save_to_gfs']:
             setattr(self, _k, save_params.get(_k, DEFAULT_SAVE_PARAMS[_k]))
 
-        for _k in ['do_restore', 'load_param_dict', 'load_step', 'from_ckpt']:
+        for _k in ['do_restore']:
```

Comment:
We need to:

1.  do a careful diff rel. master and make sure nothing else like this has escaped our attention

2. the initialize fn has to really be cleaned up.   There are three needs:

    - from_ckpt, which will load from a file as opposed to from the DB.  This should be done with the same API as @anayebi wrote but should be cleaned up to ensure that it is orthogonal from the next few issues. 
    - var_list, which will specify which things from the loaded checkpoint to actually use as opposed to initialize newly.   this should probably be a list of strings or a regex or callable which returns strings? (but think about it)
    - mapping_dict, which says, for various names of what are loaded and used, what their new names are

All three of these issues should be totally orthogonal, so that eg. specifying the mapping shouldn't get in the way of loading other variables that are not renamed.